src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -1128,8 +1175,8 @@ class Global(var currentSettings: Settings, reporter0: Reporter)
     /** A map from compiled top-level symbols to their picklers */
     val symData = new mutable.AnyRefMap[Symbol, PickleBuffer]
 
-    private var phasec: Int  = 0   // phases completed
-    private var unitc: Int   = 0   // units completed this phase
+    private var phasec: Int = 0                                     // phases completed
+    private final val unitc: AtomicInteger = new AtomicInteger(0)   // units completed this phase
```

Comment:
I think that can be a Parallel.Counter
and AtomicInteger, with get, increment and reset
we can also add assertions on the access in the same way